app/ota_client.py

```diff
@@ -413,29 +416,31 @@ def _download_raw_file(self, url, dest_file, target_hash="", fl=""):
                 # download
                 response, digest = self._download_raw(url, ftmp)
                 if response.status_code == 200:
```

Comment:
```suggestion
                if response.status_code != 200:
                    print("download error! status code: ", response.status_code)
                    return False
```
としてしまって、早めに処理を終了してしまうのがいいと思います。

Reply:
指摘部分だけだと関数内の他の部分との整合が良くないので、関数全体を修正します